smdatatools/data_processing/input_processor.py

```diff
@@ -5,10 +5,10 @@
 
 class InputProcessor:
 
-    def convert_note(line):                                                      
+    def convert_note(line: str) -> str:                                                      
         return sub('4', '1', sub('[MKLF]', '0', line))    #replaces extra notes: M, K, L, F; replaces 4 note
 
-    def parse_sm_input(sm_file):
+    def parse_sm_input(sm_file: list[str]) -> tuple[defaultdict(list), bool]:
```

Comment:
`defaultdict` is a subcalss of the built-in `dict`, so the typing can be made more generic:
1.  is `dict[str, Any]` or even 
2. `dict[str, dict[str | list[str]] | float | int ]` to be more precise

In the future, `note_data` should be refactored to a dataclass for easier typing.